### PR TITLE
Paginator Iterator

### PR DESCRIPTION
**Pull request by tklever**
An OuterIterator to flatten Paginator objects.

Allows a developer to iterate over an entire paginated set as one loop. Not much of a use case with the ArrayAdapter, however, could be helpful with Paginator Adapters that fetch remote data.

### CODE REVIEW

File: library/Zend/Paginator/PaginatorIterator.php [hunk 1]
```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace Zend\Paginator;
+
+use OuterIterator;
+
+class PaginatorIterator implements OuterIterator
```
Comment (keradus) [inline library/Zend/Paginator/PaginatorIterator.php]: it would be cool to document class as well

Comment (froschdesign) [inline library/Zend/Paginator/PaginatorIterator.php]: @keradus 
Please report this at the [issue tracker for documentation](https://github.com/zendframework/zf2-documentation).

Comment (keradus) [inline library/Zend/Paginator/PaginatorIterator.php]: To bo honest I do not see why.
I noticed that all props and methods have their own docs, even if it's just a copy-paste from php.net.
If docs for class should be done in separate repo then, imo, it should be done by author of PR.

File: library/Zend/Paginator/PaginatorIterator.php [hunk 2]
```diff
@@ -0,0 +1,113 @@
+<?php
+
```
Comment (samsonasik) [inline library/Zend/Paginator/PaginatorIterator.php]: add license header block here : 
```
/**
 * Zend Framework (http://framework.zend.com/)
 *
 * @link      http://github.com/zendframework/zf2 for the canonical source repository
 * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
 * @license   http://framework.zend.com/license/new-bsd New BSD License
 */
```

File: tests/ZendTest/Paginator/PaginatorIteratorTest.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
+<?php
+
```
Comment (samsonasik) [inline tests/ZendTest/Paginator/PaginatorIteratorTest.php]: here too : 
```
/**
 * Zend Framework (http://framework.zend.com/)
 *
 * @link      http://github.com/zendframework/zf2 for the canonical source repository
 * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
 * @license   http://framework.zend.com/license/new-bsd New BSD License
 */
```

**Comment by tklever** [comment]
Updated to add license as requested

**Comment by Ocramius** [comment]
@tklever what is exactly the use-case? I probably missed that...

**Comment by tklever** [comment]
@Ocramius Use case (as I've used it) is flat mapping lazy-loaded data, specifically remote data. It also insulates code that needs to use / process that data from specific knowledge of Zend\Paginator.

It's also been helpful when passing paginated data to OuterIterators. If you want to want to run some kind of "MappingIterator" operation over a paginated data set, you can wrap it in this iterator first.

I realize you can set the Paginator to "-1" and get the entire data set, which makes this class somewhat redundant, but again, with remote data, often that's not ideal to grab those (potentially gigantic) data sets.

**Pull request merged**